lib/videojs-hlsjs-plugin.js

```diff
@@ -123,7 +123,9 @@ var registerSourceHandler = function (videojs) {
         }
 
         function _initHlsjs() {
-            var hlsjsConfig = tech.options_.hlsjsConfig || {};
+            var hlsjsConfigRef = tech.options_.hlsjsConfig;
+            // NOTE: Hls.js will write to the reference thus change the object for later streams
+            var hlsjsConfig = hlsjsConfigRef ? JSON.parse(JSON.stringify(hlsjsConfigRef)) : {};
```

Comment:
Are you sure everything is compatible with this way of doing deep copy? Like overriding the loaders?

Reply:
Well it's pre-wrapper so it doesn't affect us but otherwise we could have the bug when we replay the video (I think it was the 2 types of `liveSync` delay config being defined at the same time or something)

Otherwise we would need a DFS cloner to be able to have classes there...

Reply:
Well can't we do what we do with other plugins? I don't think we need deep copy here.

Reply:
Well in JW we have lodash and otherwise `Object.assign` will potentially need a polyfill (or `babel-runtime`), Anyway I just replace the logic with an `Object.assign` implementation.